File: cp_rust/src/bin/abc392-b-who-is-missing?.rs
// Created by Ayush Biswas at 2026/07/08 19:52
// https://atcoder.jp/contests/adt_medium_20250521_3/tasks/abc392_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;
use std::collections::HashSet;

sol! {
    fn solution(
        [n, m]: [usize; 2],
        a: [usize]
    ) -> Lines<Words<usize>> {
        let a: HashSet<usize> = HashSet::from_iter(a.into_iter());
        let res = (1..=n).filter(|i| !a.contains(i)).collect_vec();
        vec![
            vec![res.len()].into(),
            res.into()
        ].into()
    }
}
// @code end

File: cp_nim/src/bin/abc392-b-who-is-missing?.nim
# Created by Ayush Biswas at 2026/07/19 09:44
# https://atcoder.jp/contests/adt_medium_20250521_3/tasks/abc392_b
import ../lib/cpio

# @code begin

import sequtils

var r = newReader()

let
  (n, m) = r[int, int]
  a = r[seq[int]]

let res = (1 .. n).toSeq.filterIt(it notin a)

echo res.len
echo Words(res)

# @code end
